src/machine/machine_atmega2560.go

```diff
@@ -127,6 +128,799 @@ func (p Pin) getPortMask() (*volatile.Register8, uint8) {
+			avr.TCCR3A.SetBits(avr.TCCR3A_COM3B1)
+			return 1, nil
+		case PE5: //channel C
+			avr.TCCR3A.SetBits(avr.TCCR3A_COM3C1)
+			return 2, nil
+		}
+	case 4:
+		switch pin {
+		case PH3: // channel A
+			avr.TCCR4A.SetBits(avr.TCCR4A_COM4A1)
+			return 0, nil
+		case PH4: //channel B
+			avr.TCCR4A.SetBits(avr.TCCR4A_COM4B1)
+			return 1, nil
+		case PH5: //channel C
+			avr.TCCR4A.SetBits(avr.TCCR4A_COM4C1)
+			return 2, nil
+		}
+	case 5:
+		switch pin {
+		case PL3: // channel A
+			avr.TCCR5A.SetBits(avr.TCCR5A_COM5A1)
+			return 0, nil
+		case PL4: //channel B
+			avr.TCCR5A.SetBits(avr.TCCR5A_COM5B1)
+			return 1, nil
+		case PL5: //channel C
+			avr.TCCR5A.SetBits(avr.TCCR5A_COM5C1)
+			return 2, nil
+		}
+	}
+	return 0, ErrInvalidOutputPin
+}
+
+// SetInverting sets whether to invert the output of this channel.
+// Without inverting, a 25% duty cycle would mean the output is high for 25% of
+// the time and low for the rest. Inverting flips the output as if a NOT gate
+// was placed at the output, meaning that the output would be 25% low and 75%
+// high with a duty cycle of 25%.
+//
+// Note: the invert state may not be applied on the AVR until the next call to
+// ch.Set().
+func (pwm PWM) SetInverting(channel uint8, inverting bool) {
+	switch pwm.num {
+	case 0:
+		switch channel {
+		case 0: // channel A, PB7
+			if inverting {
+				avr.PORTB.SetBits(1 << 7) // PB7 high
+				avr.TCCR0A.SetBits(avr.TCCR0A_COM0A0)
+			} else {
+				avr.PORTB.ClearBits(1 << 7) // PB7 low
+				avr.TCCR0A.ClearBits(avr.TCCR0A_COM0A0)
+			}
+		case 1: // channel B, PG5
+			if inverting {
+				avr.PORTG.SetBits(1 << 5) // PG5 high
+				avr.TCCR0A.SetBits(avr.TCCR0A_COM0B0)
+			} else {
+				avr.PORTG.ClearBits(1 << 5) // PG5 low
+				avr.TCCR0A.ClearBits(avr.TCCR0A_COM0B0)
+			}
+		}
+	case 1:
+		// Note: the COM1A0/COM1B0 bit is not set with the configuration below.
+		// It will be set the following call to Set(), however.
+		switch channel {
+		case 0: // channel A, PB5
+			if inverting {
+				avr.PORTB.SetBits(1 << 5) // PB5 high
+			} else {
+				avr.PORTB.ClearBits(1 << 5) // PB5 low
+			}
+		case 1: // channel B, PB6
+			if inverting {
+				avr.PORTB.SetBits(1 << 6) // PB6 high
+			} else {
+				avr.PORTB.ClearBits(1 << 6) // PB6 low
+			}
+		}
+	case 2:
+		switch channel {
+		case 0: // channel A, PB4
+			if inverting {
+				avr.PORTB.SetBits(1 << 4) // PB4 high
+				avr.TCCR2A.SetBits(avr.TCCR2A_COM2A0)
+			} else {
+				avr.PORTB.ClearBits(1 << 4) // PB4 low
+				avr.TCCR2A.ClearBits(avr.TCCR2A_COM2A0)
+			}
+		case 1: // channel B, PH6
+			if inverting {
+				avr.PORTH.SetBits(1 << 6) // PH6 high
+				avr.TCCR2A.SetBits(avr.TCCR2A_COM2B0)
+			} else {
+				avr.PORTH.ClearBits(1 << 6) // PH6 low
+				avr.TCCR2A.ClearBits(avr.TCCR2A_COM2B0)
+			}
+		}
+	case 3:
+		// Note: the COM3A0/COM3B0 bit is not set with the configuration below.
+		// It will be set the following call to Set(), however.
+		switch channel {
+		case 0: // channel A, PE3
```

Comment:
Maybe this switch can be collapsed a bit
```go
case 3:
    // Note: the COM3A0/COM3B0 bit is not set with the configuration below.
    if inverting {
        avr.PORTE.SetBits(1<<(channel+3))
    } else {
         avr.PORTE.ClearBits(1<<(channel+3))
    }
```
If I'm not mistaken, in which case the ones below can probably be also collapsed a tad bit.

Reply:
Nice catch! Thanks :) 
As this code is a copy/past from the atmega1280, is there a way to share the same code for both targets ? 
I don't see any build tag that would match both of them.

Reply:
They are two separate processors so they have different build tags for now. Maybe we could move the pwm code to a file called `machine_atmega_1280_1560.go` and add both builds tags like so `// +build avr,atmega2560 avr,atmega1280`